### DISCUSSION TITLE
Can we have a closing keyword to mark a PR as manually merged?

### DISCUSSION THREAD

**Discussion by eli-schwartz** [community/community#53692] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

For various reasons, some projects have a policy to merge PRs via methods other than the GitHub UI. Here are some examples:
- project is maintained on a software forge other than GitHub, and mirrored to GitHub. PRs can be submitted to GitHub too, but must be merged on the other software forge and automatically synced to GitHub
- project policy is to add additional metadata to each commit message, for example:
  - signed-off-by
  - reference to the PR that it is associated with
- requirement to cryptographically sign each commit with a private key held on a developer's machine
- developer wishes to reword the commit message to improve its descriptiveness
- developer wishes to squash or reorder commits, for example to handle commits created with ` git commit --fixup=xxx`

There are surely other reasons too.

Here are some example projects that try to automate this:
- https://github.com/leahneukirchen/git-merge-pr
- https://github.com/projg2/pram

The problem is that all these methods mean that PRs are marked in red as "closed" rather than "merged", so it's impossible to sort or filter them to see which PRs were accepted vs. rejected.

A solution would be to permit a closing keyword, e.g. `Merges: #123` in the commit message body. When present, this doesn't simply close the PR, rather it updates the PR status to indicate that it has been merged and shows the "USER merged commit xxxx into BRANCH" event.

This could either be taken on faith -- if the project claims the PR has been merged, the GitHub UI simply assumes this is honest -- or validated by comparing the commit that has this closing keyword with the prospective merge commit that GitHub *could* create if merging the PR via the GitHub UI. If both commits have identical trees, then the manually merged version has identical results and the only difference is externally handled commit message rewriting/cryptographic signing/commit reordering.

**Comment by jmarshall** [comment]
See also #6414, which you probably want to vote up.
